src/main/java/org/spongepowered/common/mixin/core/world/ExplosionMixin.java

```diff
@@ -116,22 +118,24 @@ private void onConstructed(final net.minecraft.world.World world, final Entity e
     public void doExplosionA() {
         // Sponge Start - If the explosion should not break blocks, don't bother calculating it
         if (this.impl$shouldBreakBlocks) {
-            // Sponge End
             final Set<BlockPos> set = Sets.<BlockPos>newHashSet();
-            final int i = 16;
-
-            for (int j = 0; j < 16; ++j) {
-                for (int k = 0; k < 16; ++k) {
-                    for (int l = 0; l < 16; ++l) {
-                        if (j == 0 || j == 15 || k == 0 || k == 15 || l == 0 || l == 15) {
-                            double d0 = (double) ((float) j / 15.0F * 2.0F - 1.0F);
-                            double d1 = (double) ((float) k / 15.0F * 2.0F - 1.0F);
-                            double d2 = (double) ((float) l / 15.0F * 2.0F - 1.0F);
+            final float impl$random = this.world.rand.nextFloat();
+
+            for (int j = 0; j < impl$boundingBoxSize; ++j) {
+                for (int k = 0; k < impl$boundingBoxSize; ++k) {
+                    for (int l = 0; l < impl$boundingBoxSize; ++l) {
+                        if (j == 0 || j == impl$boundingBoxSize - 1 || k == 0 || k == impl$boundingBoxSize - 1 || l == 0 || l == impl$boundingBoxSize - 1) {
+                            double d0 = (double) ((float) j / (float)(impl$boundingBoxSize - 1) * 2.0F - 1.0F);
+                            double d1 = (double) ((float) k / (float)(impl$boundingBoxSize - 1) * 2.0F - 1.0F);
+                            double d2 = (double) ((float) l / (float)(impl$boundingBoxSize - 1) * 2.0F - 1.0F);
                             final double d3 = Math.sqrt(d0 * d0 + d1 * d1 + d2 * d2);
                             d0 = d0 / d3;
                             d1 = d1 / d3;
                             d2 = d2 / d3;
-                            float f = this.size * (0.7F + this.world.rand.nextFloat() * 0.6F);
+                            float f = this.size * (0.7F + impl$random * 0.6F);
+                            if (!impl$reducedRandomness)
```

Comment:
Make sure any if statements (or any control blocks) are wrapped in braces please. We don't want Sponge to reprise [goto fail;](https://www.imperialviolet.org/2014/02/22/applebug.html)...